src/fidasim.f90

```diff
@@ -7432,19 +7463,29 @@ subroutine halo
                         !! Weight CX rates by ion source density
                         states = rates*plasma%denp
 
+                        allocate(idx_arr(3,ncell),denn_arr(nlevs,ncell))
```

Comment:
How much is the overhead caused by allocating/deallocating memory? Would it be better to pre-allocate the memory and reuse it. The `ParticleTrack` type already stores the indices we could add a `dens` variable to that as well.

Reply:
Trivial... not worth worrying about

Reply:
Clarification:
Not against augmenting the ParticleTrack type, if it logically makes sense.
Would not help speed wise, but I am all for cleaner code.